nanopub/publication.py

```diff
@@ -51,50 +54,24 @@ def _replace_blank_nodes(dummy_namespace, rdf):
         for s, p, o in rdf:
             rdf.remove((s, p, o))
             if isinstance(s, rdflib.term.BNode):
-                s = dummy_namespace[str(s)]
+                s = DUMMY_NAMESPACE[str(s)]
             if isinstance(o, rdflib.term.BNode):
-                o = dummy_namespace[str(o)]
+                o = DUMMY_NAMESPACE[str(o)]
             rdf.add((s, p, o))
 
-    @classmethod
-    def from_assertion(cls, assertion_rdf: rdflib.Graph,
-                       introduces_concept: rdflib.term.BNode = None,
-                       derived_from=None, assertion_attributed_to=None,
-                       attribute_assertion_to_profile: bool = False,
-                       provenance_rdf: rdflib.Graph = None,
-                       pubinfo_rdf: rdflib.Graph = None
-                       ):
-        """
-        Construct Nanopub object based on given assertion. Any blank nodes in the rdf graph are
-        replaced with the nanopub's URI, with the blank node name as a fragment. For example, if
-        the blank node is called 'step', that would result in a URI composed of the nanopub's (base)
-        URI, followed by #step.
+    @staticmethod
+    def _validate_from_assertion_arguments(
+            introduces_concept: rdflib.term.BNode, derived_from, assertion_attributed_to,
+            attribute_assertion_to_profile: bool, provenance_rdf: rdflib.Graph,
+            pubinfo_rdf: rdflib.Graph):
 
```

Comment:
Should this not also have a docstring? (even though it is indeed an "internal" method)

Reply:
I've added one:         'Validate arguments for `from_assertion` method.' 
I usually don't add them if they are literally the same as the name of a function. But it's good practice anyway.